pilot/pkg/model/cluster_local.go

```diff
@@ -64,14 +64,14 @@ func NewClusterLocalProvider(e *Environment) ClusterLocalProvider {
 var _ ClusterLocalProvider = &clusterLocalProvider{}
 
 type clusterLocalProvider struct {
-	mutex sync.Mutex
+	mutex sync.RWMutex
```

Comment:
could just use atomic.Pointer here?

Reply:
It should be possible to use atomic.Pointer here, but I prefer to merge this pr first, and then I submit a pr to handle all syncs that can be replaced by atomic.Pointer

Reply:
for most cases it is recommended to use mutex